app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :favorite, :unfavorite]
+  before_action :block_unauthorized_changes, only: [:edit, :update, :destroy]
   before_filter :authenticate_user!
   # GET /posts
-  # GET /posts.json
```

Comment:
It looks like you generated this with a scaffold, which is okay, but make sure you clean up after it. These comments can be removed.